smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/utils/AuthUtils.kt

```diff
@@ -0,0 +1,39 @@
+package software.amazon.smithy.swift.codegen.utils
```

Comment:
This is a refactor for how we codegen the list of auth schemes supported by the service. Before, there was no auth scheme in `smithy-swift`, and so in `smithy-swift` codegen, the value of `authSchemes` field was always an empty array. The `getModeledAuthSchemesSupportedBySDK` was in `aws-sdk-swift` and only concerned itself with sigv4 & sigv4a. But now that `BearerTokenAuthScheme` is added to `smithy-swift`, generating list of auth schemes was pulled into `smithy-swift`.